doc/neps/nep-0031-uarray.rst

```diff
@@ -39,41 +47,51 @@ functions that can be easily expressed in terms of others, as well as a reposito
 implementations.
 
 This NEP proposes the following: That ``unumpy`` `[8]`_  becomes the recommended override mechanism
-for the NumPy API.
+for the parts of the NumPy API not yet covered by ``__array_function__`` or ``__array_ufunc__``,
+and that ``uarray`` is vendored into a new namespace within NumPy to give users and downstream dependencies
+access to these overrides.  This vendoring mechanism is similar to what SciPy decided to do for
+making ``scipy.fft`` overridable (see `[10]`_).
 
 
 Detailed description
 --------------------
 
-This section will not attempt to explain the specifics or the mechanism of ``uarray``,
-that is explained in the ``uarray`` documentation. `[1]`_ However, the NumPy community
-will have input into the design of ``uarray``, and any backward-incompatible changes
-will be discussed on the mailing list.
+_Note that this section will not attempt to explain the specifics or the mechanism of ``uarray``,_
+_that is explained in the ``uarray`` documentation. `[1]`_ However, the NumPy community_
+_will have input into the design of ``uarray``, and any backward-incompatible changes_
+_will be discussed on the mailing list._
+
+The way we propose the overrides will be used by end users is::
+
+    from numpy import unumpy
+    TODO
 
-The first goal of this NEP is as follows: To complete an overridable version of NumPy,
-called ``unumpy`` `[8]`_, the implementation of which is already underway. Again, ``unumpy``
-will not be explained here, the reader should refer to its documentation for this purpose.
+And a library that implements a NumPy-like API will use it like::
+
+    TODO: example corresponding to NEP 30 `duckarray`
 
 The only change this NEP proposes at its acceptance, is to make ``unumpy`` the officially recommended
-way to override NumPy. ``unumpy`` will remain a separate repository/package, and will be developed
+way to override NumPy. ``unumpy`` will remain a separate repository/package (which we propose to vendor
+to avoid a hard dependency, and use the separate ``unumpy`` package only if it is installed)
+rather than depend on for the time being), and will be developed
 primarily with the input of duck-array authors and secondarily, custom dtype authors, via the usual
 GitHub workflow. There are a few reasons for this:
 
 * Faster iteration in the case of bugs or issues.
 * Faster design changes, in the case of needed functionality.
-* Lower likelihood to be stuck with a bad design decision.
+* ``unumpy`` will work with older versions of NumPy as well.
 * The user and library author opt-in to the override process,
   rather than breakages happening when it is least expected.
   In simple terms, bugs in ``unumpy`` mean that ``numpy`` remains
   unaffected.
-* The upgrade pathway to NumPy 2.0 becomes simpler, requiring just
-  a backend change, and allowing both to exist side-by-side.
+
+**FIXME: this section doesn't match the proposal. in the abstract and motivation anymore.**
```

Comment:
@rgommers Can you expand on this a little bit?

> obviates the need to add an ever-growing list of new protocols for each new type of function or object that needs to become overridable.

> It assumes that there are parts of the API that need to be overridable, and that these will grow over time. It provides a general framework and a mechanism to overridable, and that these will grow over time.

So it's consistent, even though it explains other use-cases. Would you like for this section to be removed? Personally, I'd at least like to mention the possibility of having such a system, and for `unumpy` as an alternative to other overrides.

Reply:
> Personally, I'd at least like to mention the possibility of having such a system, and for unumpy as an alternative to other overrides.

Yes, I think it's good to mention that. 

> Can you expand on this a little bit?

If the whole thing is opt-in and kept in a separate namespace, as in the abstract and motivation sections, then a lot of this reads strangely. The examples don't work as `np.<funcname>`, because that assumes you've already completely taken over the whole namespace. It's clearer when you add the full example, so I'll comment on your new PR.